src/TLuaInterpreter.cpp

```diff
@@ -11830,35 +11830,288 @@ int TLuaInterpreter::sendSocket( lua_State * L )
+            lua_pushnil(L);
+            lua_pushfstring(L, "bad argument #1 value (hostname must not be empty!)");
+            return 2;
+        }
+    }
+    if (!lua_isnoneornil(L, 2)) {
+        if (!lua_isnumber(L, 2)) {
+            lua_pushfstring(L, "ircSaveServer: bad argument #2 type (port number as number is optional {default = 6667}, got %s!)", lua_typename(L, 2));
+            return lua_error(L);
+        } else {
+            port = lua_tointeger(L, 2);
+            if (port > 65535 || port < 1) {
+                lua_pushnil(L);
+                lua_pushfstring(L, "invalid port number %d given, if supplied it must be in range 1 to 65535, {defaults to 6667 if not provided}", port);
+                return 2;
+            }
+        }
+    }
+
+    Host* pHost = &getHostFromLua(L);
+    QPair<bool, QString> result = dlgIRC::writeIrcHostName(pHost, QString::fromStdString(addr));
+    if (!result.first) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "unable to save hostname, reason: %s", result.second.toUtf8().constData());
+        return 2;
+    }
+
+    result = dlgIRC::writeIrcHostPort(pHost, port);
+    if (!result.first) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "unable to save port, reason: %s", result.second.toUtf8().constData());
+        return 2;
+    }
+
+    lua_pushboolean(L, true);
+    lua_pushnil(L);
+    return 2;
+}
+
+/** ircSaveChannels( channels )
+ *  Updates IRC client auto-join channels configuration value.
+ *  Channels must a table containing at least one valid channel name string.
+ *  Does not apply changes to active client until ircRestart() is called.
+ */
+int TLuaInterpreter::ircSaveChannels(lua_State* L)
+{
+    QStringList newchannels;
+    if (!lua_istable(L, 1) )
     {
-        lua_pushstring( L, "sendIrc: wrong argument type" );
-        lua_error( L );
-        return 1;
+        lua_pushfstring(L, "ircSaveChannels: bad argument #1 type (channels as table expected, got %s!)", lua_typename(L, lua_type(L, 1)));
+        return lua_error(L);
     }
     else
     {
-        text = lua_tostring( L, 2 );
+        lua_pushnil( L );
+        while( lua_next( L, 1 ) != 0 )
+        {
+            // key at index -2 and value at index -1
+            if( lua_type(L, -1) == LUA_TSTRING )
+            {
+                QString c = lua_tostring( L, -1 );
+                if (!c.isEmpty() && (c.startsWith("#") || c.startsWith("&") || c.startsWith("+")) ) {
+                    newchannels << c;
+                }
+            }
+            lua_pop(L, 1);
+        }
     }
-    QString chan = who.c_str();
-    QString txt = text.c_str();
-    if( ! mudlet::self()->mpIRC ) return 0;
-    mudlet::self()->mpIRC->connection->sendCommand( IrcCommand::createMessage( chan, txt ) );
-    return 0;
+
+    if (newchannels.count() == 0) {
+        lua_pushnil(L);
+        lua_pushstring(L, "bad argument #1 value (channels must contain at least 1 valid channel name!)");
```

Comment:
Using `lua_error()` - which _aborts_ the execution of the item of lua code and will allow a string at the top of the lua stack to be used as an error message which the caller can read... We usually use it for a case where the user has not provided a suitable type of argument as that tends to mean they are misusing the API and that they should get a "you can't do _that_ with _those_" {arguments} message instead of actually trying to do anything.